### PR TITLE
fix shellcheck failures list-resources.sh

### PR DESCRIPTION
**Pull request by xichengliudui**
/kind cleanup
/priority backlog
/kind cleanup
/sig testing

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/list-resources.sh [hunk 1]
```diff
@@ -49,20 +49,20 @@ function gcloud-list() {
...
+    if result=$(gcloud "${group}" "${resource}" list --project="${PROJECT}" "${filter:+--filter="$filter"}" "${@:4}"); then
+      if [[ -n "${GREP_REGEX}" ]]; then
         result=$(echo "${result}" | grep "${GREP_REGEX}" || true)
       fi
       echo "${result}"
       return
     fi
     echo -e "Attempt ${attempt} failed to list ${resource}. Retrying." >&2
-    attempt=$(($attempt+1))
+    attempt=$(attempt + 1)
```
Comment (fejta) [inline cluster/gce/list-resources.sh]: I don't think this works

Comment (xichengliudui) [reply cluster/gce/list-resources.sh]: has been updated.

File: cluster/gce/list-resources.sh [hunk 2]
```diff
@@ -49,20 +49,20 @@ function gcloud-list() {
   local attempt=1
   local result=""
   while true; do
-    if result=$(gcloud ${group} ${resource} list --project=${PROJECT} ${filter:+--filter="$filter"} ${@:4}); then
-      if [[ ! -z "${GREP_REGEX}" ]]; then
+    if result=$(gcloud "${group}" "${resource}" list --project="${PROJECT}" "${filter:+--filter="$filter"}" "${@:4}"); then
+      if [[ -n "${GREP_REGEX}" ]]; then
         result=$(echo "${result}" | grep "${GREP_REGEX}" || true)
```
Comment (fejta) [inline cluster/gce/list-resources.sh]: should it be grep -E?

**Review by fejta — COMMENTED** [review]
/lgtm
/approve

File: cluster/gce/list-resources.sh (hunk 2 above)
Comment (xichengliudui) [reply cluster/gce/list-resources.sh]: /assign @yujuhong , what do you think?

Comment (yujuhong) [reply cluster/gce/list-resources.sh]: `grep` and `grep -E` supports different syntax. I'd suggest not change this in a cleanup PR.

**Comment by yujuhong** [comment]
/approve
You need to check resolve the conflict

**Comment by xichengliudui** [comment]
Resolve file conflicts `need-rebase`.

File: cluster/gce/list-resources.sh [hunk 3]
```diff
@@ -49,20 +49,20 @@ function gcloud-list() {
   local attempt=1
   local result=""
   while true; do
-    if result=$(gcloud ${group} ${resource} list --project=${PROJECT} ${filter:+--filter="$filter"} ${@:4}); then
-      if [[ ! -z "${GREP_REGEX}" ]]; then
+    if result=$(gcloud "${group}" "${resource}" list --project="${PROJECT}" "${filter:+--filter="$filter"}" "${@:4}"); then
+      if [[ -n "${GREP_REGEX}" ]]; then
```
Comment (cblecker) [inline cluster/gce/list-resources.sh]:
```suggestion
      if [[ -n "${GREP_REGEX:-}" ]]; then
```

Comment (xichengliudui) [reply cluster/gce/list-resources.sh]: has been updated.

**Pull request merged**